docs/desktop-wpf/data/data-binding-overview.md

```diff
@@ -0,0 +1,434 @@
+author: thraka
+ms.date: 09/19/2019
+ms.author: adegeo
+dev_langs: ["csharp", "vb"]
+---
+
+# Data Binding Overview
+
+Windows Presentation Foundation (WPF) data binding provides a simple and consistent way for applications to present and interact with data. Elements can be bound to data from a variety of data sources in the form of common language runtime (CLR) objects and XML. Any <xref:System.Windows.Controls.ContentControl> such as <xref:System.Windows.Controls.Button> and any <xref:System.Windows.Controls.ItemsControl> such as <xref:System.Windows.Controls.ListBox> and <xref:System.Windows.Controls.ListView>, have built-in functionality to enable flexible styling of single data items or collections of data items. Sort, filter, and group views can be generated on top of the data.
+
+The data binding functionality in WPF has several advantages over traditional models, including a broad range of properties that inherently support data binding, flexible UI representation of data, and clean separation of business logic from UI.
+
+This article first discusses concepts fundamental to WPF data binding and then goes into the usage of the <xref:System.Windows.Data.Binding> class and other features of data binding.
+
+[!INCLUDE [desktop guide under construction](../../../includes/desktop-guide-preview-note.md)]
+
+## What is data binding
+
+Data binding is the process that establishes a connection between the application UI and business logic. If the binding has the correct settings and the data provides the proper notifications, then, when the data changes its value, the elements that are bound to the data reflect changes automatically. Data binding can also mean that if an outer representation of the data in an element changes, then the underlying data can be automatically updated to reflect the change. For example, if the user edits the value in a `TextBox` element, the underlying data value is automatically updated to reflect that change.
+
+A typical use of data binding is to place server or local configuration data into forms or other UI controls. In WPF, this concept is expanded to include the binding of a broad range of properties to a variety of data sources. In WPF, dependency properties of elements can be bound to CLR objects (including ADO.NET objects or objects associated with Web Services and Web properties) and XML data.
+
+For an example of data binding, take a look at the following application UI from the [Data Binding Demo][data-binding-demo]:
+
+![Data binding sample screenshot](./media/data-binding-overview/demo.png "DataBinding_DataBindingDemo")
+
+The above is the UI of an application that displays a list of auction items. The application demonstrates the following features of data binding:
+
+- The content of the ListBox is bound to a collection of *AuctionItem* objects. An *AuctionItem* object has properties such as *Description*, *StartPrice*, *StartDate*, *Category*, *SpecialFeatures*, etc.
```

Comment:
@mairaw @rpetrusha 

I agree. I've done a lot of clean up while porting this article but there is more to do. One problem I'm having is that there are so many xref links that it makes it look ugly when you read through it in published mode. Further down it gets messier. One paragraph (especially [the validation section below](https://github.com/dotnet/docs/pull/13780/files#diff-95682880d8470fb87de4ba5e56194f26R413) can easily have 10 xref links.

Reply:
Links behaved differently in MSDN so people used "xref" links more freely back then. It's ok not to link to the API every single time. But if you're not linking, the name should be fenced to prevent localization.